test/e2e/storage/testsuites/base.go

```diff
@@ -308,15 +308,42 @@ func (r *VolumeResource) CleanupResource() error {
 					r.Pv.Name, v1.PersistentVolumeReclaimDelete)
 			}
 			if r.Pvc != nil {
+				cs := f.ClientSet
+				pv := r.Pv
+				if pv == nil && r.Pvc.Name != "" {
+					// This happens for late binding. Check whether we have a volume now that we need to wait for.
+					pvc, err := cs.CoreV1().PersistentVolumeClaims(r.Pvc.Namespace).Get(context.TODO(), r.Pvc.Name, metav1.GetOptions{})
+					switch {
+					case err == nil:
+						if pvc.Spec.VolumeName != "" {
+							pv2, err := cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+							switch {
+							case err == nil:
+								pv = pv2
```

Comment:
Do we need a pv2? pv is supposed to be nil here.

Reply:
I've been bitten by scoping rules in Go before, so I prefer being explicit about updating some variable in an outer scope.

The "simple" approach doesn't work:
```
pv, err := cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
```

It creates a new `pv` variable which shadows the one we really want to update.

This works, but I find it ugly:
```
var err error
pv, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
```

:shrug:

Reply:
`err` is already initialized at L315, so you shouldn't need to declare it here?

Reply:
That would work... until that L315 changes.

I still don't like how scoping works here, but as it reduces the number of lines I'll go with it:
```
	pv, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
	if err != nil {
		cleanUpErrs = append(cleanUpErrs, errors.Wrapf(err, "Failed to find PV %v", pvc.Spec.VolumeName))
	}
```

This simplified version also assumes that pv is nil in case of an error and not some incomplete garbage. Unfortunately (IMHO) Go doesn't enforce that.